apps/frontend/src/components/common/proposalFilters/TechniqueFilter.tsx

```diff
@@ -0,0 +1,124 @@
+import Box from '@mui/material/Box';
+import Divider from '@mui/material/Divider';
+import FormControl from '@mui/material/FormControl';
+import InputLabel from '@mui/material/InputLabel';
+import ListSubheader from '@mui/material/ListSubheader';
+import MenuItem from '@mui/material/MenuItem';
+import Select from '@mui/material/Select';
+import React, { Dispatch } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useSearchParams } from 'react-router-dom';
+
+import { TechniqueFilterInput, TechniqueFragment } from 'generated/sdk';
+
+export enum TechniqueFilterEnum {
+  ALL = 'all',
+  MULTI = 'multi',
+}
+
+type TechniqueFilterProps = {
+  techniques?: TechniqueFragment[];
+  isLoading?: boolean;
+  onChange?: Dispatch<TechniqueFilterInput>;
+  shouldShowAll?: boolean;
+  shouldShowMultiple?: boolean;
+  showMultiTechniqueProposals?: boolean;
+  techniqueId?: number | null;
+};
+
+const TechniqueFilter = ({
+  techniques,
+  isLoading,
+  techniqueId,
+  onChange,
+  shouldShowAll,
+  shouldShowMultiple,
+  showMultiTechniqueProposals,
+}: TechniqueFilterProps) => {
+  const [, setSearchParams] = useSearchParams();
+  const { t } = useTranslation();
+
+  if (techniques === undefined) {
+    return null;
+  }
+
+  /**
+   * NOTE: We might use https://material-ui.com/components/autocomplete/.
+   * If we have lot of dropdown options to be able to search.
+   */
```

Comment:
Is these relevant in the current context if it is we can use autocomplete then know its sorted.